src/components/HOC/AuthRedirect.tsx

```diff
@@ -0,0 +1,21 @@
+import Router from "next/router";
+import useUserInfo from "../../hooks/useUserInfo";
+
+const withAuthRedirect = (Component) => {
+  const route = (props) => {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const { isLoggedIn } = useUserInfo();
+
+    if (isLoggedIn) {
+      return <Component {...props} />;
+    }
+
+    typeof window !== "undefined" && Router.push("/auth");
```

Comment:
This logic is flawed currently. Firstly, this typeof line is very questionable. Use `if`. Also currently when the user is not logged in and is SSR, the content is still rendered. Return `null` if it is SSR and user is not logged in,